File: frontend/src/types/monitoring.ts
export interface SystemMetric {
  id: string;
  metric_name: string;
  metric_value: number;
  metric_unit?: string;
  tags?: Record<string, any>;
  timestamp: string;
}

export interface SystemMetricCreate {
  metric_name: string;
  metric_value: number;
  metric_unit?: string;
  tags?: Record<string, any>;
}

export interface AuditLog {
  id: string;
  user_id?: string;
  action: string;
  resource_type?: string;
  resource_id?: string;
  details?: Record<string, any>;
  ip_address?: string;
  user_agent?: string;
  timestamp: string;
}

export interface AuditLogCreate {
  user_id?: string;
  action: string;
  resource_type?: string;
  resource_id?: string;
  details?: Record<string, any>;
  ip_address?: string;
  user_agent?: string;
}

export interface OperationLog {
  id: string;
  operation_type: string;
  operation_data: Record<string, any>;
  status: string;
  error_message?: string;
  execution_time?: number;
  timestamp: string;
}

export interface OperationLogCreate {
  operation_type: string;
  operation_data: Record<string, any>;
  status: string;
  error_message?: string;
  execution_time?: number;
}

export interface SystemStats {
  cpu_usage: number;
  memory_usage: number;
  disk_usage: number;
  network_rx: number;
  network_tx: number;
  active_connections: number;
  timestamp: string;
}

export interface ServiceStatus {
  service_name: string;
  status: string;
  uptime?: number;
  last_check: string;
}

export interface AlertRule {
  id: string;
  name: string;
  metric_name: string;
  threshold: number;
  operator: string;
  severity: string;
  is_enabled: boolean;
  created_at: string;
}

export interface Alert {
  id: string;
  rule_id: string;
  message: string;
  severity: string;
  status: string;
  created_at: string;
  resolved_at?: string;
}

export interface LogQuery {
  start_time?: string;
  end_time?: string;
  level?: string;
  service?: string;
  message?: string;
  limit: number;
  offset: number;
}

export interface LogResponse {
  logs: Record<string, any>[];
  total: number;
  has_more: boolean;
}

File: backend/app/schemas/monitoring.py
from datetime import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, field_validator
from decimal import Decimal

# System Metric Schemas
class SystemMetricBase(BaseModel):
    metric_name: str
    metric_value: Decimal
    metric_unit: Optional[str] = None
    tags: Optional[Dict[str, Any]] = None

class SystemMetricCreate(SystemMetricBase):
    pass

class SystemMetric(SystemMetricBase):
    id: int
    timestamp: datetime

    class Config:
        from_attributes = True

# Audit Log Schemas
class AuditLogBase(BaseModel):
    user_id: Optional[int] = None
    action: str
    resource_type: Optional[str] = None
    resource_id: Optional[int] = None
    details: Optional[Dict[str, Any]] = None
    ip_address: Optional[str] = None
    user_agent: Optional[str] = None

class AuditLogCreate(AuditLogBase):
    pass

class AuditLog(AuditLogBase):
    id: int
    timestamp: datetime

    class Config:
        from_attributes = True

# Operation Log Schemas
class OperationLogBase(BaseModel):
    operation_type: str
    operation_data: Dict[str, Any]
    status: str  # 'success', 'failed', 'in_progress'
    error_message: Optional[str] = None
    execution_time: Optional[int] = None  # in milliseconds

    @field_validator('status')
    @classmethod
    def validate_status(cls, v):
        allowed_statuses = ['success', 'failed', 'in_progress']
        if v not in allowed_statuses:
            raise ValueError(f'状态必须是: {", ".join(allowed_statuses)}')
        return v

class OperationLogCreate(OperationLogBase):
    pass

class OperationLog(OperationLogBase):
    id: int
    timestamp: datetime

    class Config:
        from_attributes = True

# Monitoring Dashboard Schemas
class SystemStats(BaseModel):
    cpu_usage: float
    memory_usage: float
    disk_usage: float
    network_rx: int
    network_tx: int
    active_connections: int
    timestamp: datetime

class ServiceStatus(BaseModel):
    service_name: str
    status: str  # 'running', 'stopped', 'error'
    uptime: Optional[int] = None
    last_check: datetime

class AlertRule(BaseModel):
    id: int
    name: str
    metric_name: str
    threshold: float
    operator: str  # 'gt', 'lt', 'eq', 'gte', 'lte'
    severity: str  # 'info', 'warning', 'error', 'critical'
    is_enabled: bool
    created_at: datetime

class Alert(BaseModel):
    id: int
    rule_id: int
    message: str
    severity: str
    status: str  # 'active', 'resolved', 'acknowledged'
    created_at: datetime
    resolved_at: Optional[datetime] = None

# Log Query Schemas
class LogQuery(BaseModel):
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    level: Optional[str] = None
    service: Optional[str] = None
    message: Optional[str] = None
    limit: int = 100
    offset: int = 0

class LogResponse(BaseModel):
    logs: List[Dict[str, Any]]
    total: int
    has_more: bool